### PR TITLE
rustdoc: Add an unstable option to print all unversioned files

### PR DESCRIPTION
**Pull request by jyn514**
This allows sharing those files between different doc invocations
without having to know their names ahead of time.

Helps with https://github.com/rust-lang/docs.rs/issues/1302.
r? @GuillaumeGomez cc @pietroalbini @Nemo157

### CODE REVIEW

**Comment by petrochenkov** [comment]
Meta: could you add "rustdoc" to PR names so that it's clear that an option is added (or some other change is made) to rustdoc rather than to rustc.
(E.g. "rustdoc: Add an unstable option to print all unversioned files".)

Labels like `T-rustdoc` are useless for filtering PRs because they are not displayed neither in mail clients, nor even in GitHubs own notification reader.

**Comment by jyn514** [comment]
Sure thing, I'll do that.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
It's kinda doing the opposite of what I had in mind but the end result is the same so it's just as good.

Can you add a test for it the newly added option please?

**Comment by jyn514** [comment]
> Can you add a test for it the newly added option please?

Oops, I forgot to commit it to git. Should be up now.

File: src/librustdoc/config.rs [hunk 1]
```diff
@@ -315,6 +315,13 @@ impl Options {
             return Err(0);
         }
 
+        if matches.opt_strs("print").iter().any(|opt| opt == "unversioned-files") {
+            for file in crate::html::render::FILES_UNVERSIONED.keys() {
+                println!("{}", file);
+            }
+            return Err(0);
```
Comment (GuillaumeGomez) [inline src/librustdoc/config.rs:322]: This is really bad API usage we have here... We should clean that up later on.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! r=me once CI pass

**Pull request merged**